### ISSUE TITLE
About the hyper-parameter settings

### ISSUE THREAD

**Issue by hengrao02** [egetaga/TimePFN#1]
I have some questions regarding the LMC_Synth.py implementation and would appreciate your guidance:

1. Regarding experimental settings:
Could you please provide more details on how to properly configure the Dirichlet and Weibull distribution parameters (dirichlet_min, dirichlet_max, weibull_shape, etc.)? What are the recommended value ranges and their impact on the synthetic data generation?

2. Concerning data size:
When using the following hyperparameters, the generated synthetic data reaches 18GB in size:
```
parser = argparse.ArgumentParser()
parser.add_argument("-N", "--num-series", type=int, default=15000)
parser.add_argument("-J", "--max-kernels", type=int, default=5)
parser.add_argument("-C", "--num-channels", type=int, default=160)
parser.add_argument("-O", "--output", type=str, default="LMC_synth_MTS.arrow")
parser.add_argument("-D", "--directory", type=str, default="./")
parser.add_argument("-L", "--length", type=int, default=1024)
parser.add_argument("-M", "--dirichlet_min", type=float, default=0.1)
parser.add_argument("-X", "--dirichlet_max", type=float, default=1.0)
parser.add_argument("-S", "--scale", type=float, default=10.0)
parser.add_argument("-W", "--weibull_shape", type=float, default=2.0)
parser.add_argument("-Z", "--weibul_scale", type=int, default=20)
```
Is this data size expected given these parameters? If not, could you suggest appropriate parameter adjustments to generate smaller datasets?

3. Model training guidance:
Could you provide instructions on how to train a few-shot TimePFN or iTransformer-PFN model?

Thank you for your time and expertise. I look forward to your response.

**Comment by egetaga** [comment]
Hi, thanks for your interest in our paper! Happy to share how to work with synthetic generation. 

* dirichlet_min/max control the range for mixture weights over kernels. Lower values push sparser mixtures (few dominant latent functions); higher values yield more uniform mixing.
* with the Weibull, we choose the number of latent functions. 

In practice, these came from a few hand-tuned searches minimizing validation loss. You can likely do better with a small grid or random search. These values were tuned for a 160-channel setup in our paper; if you scale to much higher dimensional multivariate data, expect the best settings to shift. A good workflow is:

1. visualize the number of latent functions (histogram) and sampled series to prune obviously bad regions;
2. then run a tighter search and pick by validation loss.

For training, we found a simple curriculum to work best:

* First, pretrain purely on kernel-synth data (where each channel is independently generated). This helps the model learn basic univariate forecasting structure. You can train this stage for a few hundred thousand steps with a moderate learning rate.
* Then, fine-tune on LMC-Synth data (where channels are correlated) using a smaller learning rate. This helps the model adapt to multivariate dependencies and typically speeds up convergence while improving stability.

Regarding the 18 GB data size: that’s normal and expected for these parameters. If disk space is an issue, you can generate data online (on-the-fly) through the dataloader rather than saving it all to disk.

For the few-shot settings, and the training, we have already provided the scripts. Especially few-shot settings are very straightforward following from the scripts. 

Let us know if you have any more questions!
